src/lib/io/sol-iio.c

```diff
@@ -870,6 +889,8 @@ sol_iio_close(struct sol_iio_device *device)
     SOL_NULL_CHECK(device);
 
     SOL_PTR_VECTOR_FOREACH_REVERSE_IDX (&device->channels, channel, i) {
+        /* Deactivate device firmware/hardware after use */
+        set_channel_enabled(device, channel->name, false);
```

Comment:
@edersondisouza, @bdilly, @yongli3, @barbieri 

Currently the `iio_del_channel(struct sol_iio_channel *channel) `only take one parameter. 

If want to` set_channel_enabled(false)` in `iio_del_channel()`, then need change to `iio_del_channel(sol_iio_device *device, struct sol_iio_channel *channel)` 

Because in set_channel_enabled() need the device to craft the path to deactivate 

All agree with the above changes? If yes then I will push the changes.